### PR TITLE
Adding Cgroup customization via CreateVM

### LINKED ISSUE #417: Make cgroup path configurable from CreateVM

### PR DESCRIPTION
**Pull request by xibz**
This commit allows for configuration of the cgroup path. If no value is
provided the default value of /firecracker-containerd will be used
making the full cgroup path of
/sys/fs/cgroup/cpu/firecracker-containerd/<vmID>

Fixes #417 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: proto/firecracker.proto [hunk 1]
```diff
@@ -102,5 +102,11 @@ message JailerConfig {
     string Mems = 3;
     uint32 UID = 4;
     uint32 GID = 5;
+	// CgroupPath is used to dictate where the cgroup should be located
+	// relative to the cgroup directory which is
+	// /sys/fs/cgroup/cpu/<CgroupPath>/<vmID>
```
Comment (kzys) [inline proto/firecracker.proto:107]: @ofiliz Now clients can inject an arbitrary string, but still cannot remove the last vmID part. Does it work?

File: runtime/jailer_test.go [hunk 2]
```diff
@@ -96,3 +100,39 @@ func TestJailer_invalidUIDGID(t *testing.T) {
...
 	assert.Error(t, err, "expected invalid uid and gid error, but received none")
 }
+
+func TestNewJailer_Isolated(t *testing.T) {
+	prepareIntegTest(t)
+
+	ociBundlePath, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+	defer os.RemoveAll(ociBundlePath)
+	err = exec.Command("cp", "firecracker-runc-config.json.example", filepath.Join(ociBundlePath, "config.json")).Run()
```
Comment (kzys) [inline runtime/jailer_test.go:110]: How about using CombinedOutput? If the cp doesn't work (e.g. disk is full), the err variable may not contain the information.

File: runtime/service_integ_test.go [hunk 3]
```diff
@@ -470,7 +479,7 @@ func testMultipleExecs(
 		require.NoError(t, err, "failed to stat root path of jailer")
 		_, err = os.Stat(filepath.Join("/sys/fs/cgroup/cpu", cgroupPath))
 		require.NoError(t, err, "failed to stat cgroup path of jailer")
-		assert.Equal(t, filepath.Join("/firecracker-containerd", vmIDStr), cgroupPath)
+		assert.Regexp(t, fmt.Sprintf(".?.*/%s", vmIDStr), cgroupPath)
```
Comment (kzys) [inline runtime/service_integ_test.go:482]: Would it be `fmt.Sprintf(".+/%s", vmIDStr)`?

Comment (xibz) [reply runtime/service_integ_test.go:482]: Ah yea, that simplifies it. Forgot about the + meaning one or more.

File: runtime/jailer_test.go (hunk 2 above)
Comment (xibz) [reply runtime/jailer_test.go:110]: That's fair. Will change.

File: proto/firecracker.proto (hunk 1 above)
Comment (ofiliz) [reply proto/firecracker.proto:107]: That's the same VMID in the CreateVMRequest, right? Yes, that should be fine.

**Pull request merged**